<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>672</width>
    <height>558</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="openImage">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Open Image</string>
    </property>
   </widget>
   <widget class="QLabel" name="showImgIn">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>60</y>
      <width>300</width>
      <height>300</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="showImgOut">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>60</y>
      <width>300</width>
      <height>300</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>370</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Input Image</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>370</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Output Image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="convImg">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>430</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Convolution</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="maskSize">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>430</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Size</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="maskCoefficients">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>430</y>
      <width>471</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Mask Coefficients</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="maskSizeOrder">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>480</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Size</string>
    </property>
   </widget>
   <widget class="QPushButton" name="medianImg">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>480</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Median</string>
    </property>
   </widget>
   <widget class="QPushButton" name="maxImg">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>480</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Max</string>
    </property>
   </widget>
   <widget class="QPushButton" name="minImg">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>480</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Min</string>
    </property>
   </widget>
   <widget class="QPushButton" name="gaussianImg">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>480</y>
      <width>221</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Marr-Hildreth Detector</string>
    </property>
   </widget>
   <widget class="QPushButton" name="LoGImg">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>480</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>LoG</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>672</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>openImage</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>openImg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>75</x>
     <y>57</y>
    </hint>
    <hint type="destinationlabel">
     <x>132</x>
     <y>37</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>medianImg</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>medianImg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>95</x>
     <y>511</y>
    </hint>
    <hint type="destinationlabel">
     <x>112</x>
     <y>542</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maxImg</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>maxImg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>211</x>
     <y>533</y>
    </hint>
    <hint type="destinationlabel">
     <x>248</x>
     <y>549</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>minImg</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>minImg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>278</x>
     <y>533</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>542</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>convImg</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>convImg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>584</x>
     <y>464</y>
    </hint>
    <hint type="destinationlabel">
     <x>604</x>
     <y>491</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gaussianImg</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>MarrHildrethImg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>440</x>
     <y>533</y>
    </hint>
    <hint type="destinationlabel">
     <x>155</x>
     <y>592</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LoGImg</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>LoGImg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>514</y>
    </hint>
    <hint type="destinationlabel">
     <x>426</x>
     <y>543</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openImg()</slot>
  <slot>grayImg()</slot>
  <slot>medianImg()</slot>
  <slot>maxImg()</slot>
  <slot>minImg()</slot>
  <slot>convImg()</slot>
  <slot>convImgGray()</slot>
  <slot>MarrHildrethImg()</slot>
  <slot>LoGImg()</slot>
 </slots>
</ui>
